### ISSUE TITLE
Make it easier to customize ObjectMapper in QueueMessageHandler

### ISSUE THREAD

**Issue by andrewdmay** [spring-cloud/spring-cloud-aws#193]
We make use of custom serializers/deserializers and mixins with Jackson and need to use a customized ObjectMapper with QueueMessageHandler/@SqsListener.

We have been creating a customized QueueMessageHandlerFactory:

```
    @Bean
    public QueueMessageHandlerFactory queueMessageHandlerFactory(ObjectMapper objectMapper) {
        MappingJackson2MessageConverter mappingJackson2MessageConverter = new MappingJackson2MessageConverter();
        mappingJackson2MessageConverter.setSerializedPayloadClass(String.class);
        mappingJackson2MessageConverter.setObjectMapper(objectMapper);
        QueueMessageHandlerFactory factory = new QueueMessageHandlerFactory();
        factory.setArgumentResolvers(Arrays.asList(new PayloadArgumentResolver(mappingJackson2MessageConverter)));
        return factory;
    }
```

However while this works fine when the listener only requires the payload it does not work with the `@Header/@Headers` annotations because the PayloadArgumentResolver is added as a customer argument resolver before the standard resolvers defined in `QueueMessageHandler.initArgumentResolvers`.

I could add the header resolvers to the customer resolvers before the payload resolver, but that does not seem like an ideal solution.

Ideally if there was an `ObjectMapper` in the spring context it would be preferable if `SqsConfiguration` used it to configure the `QueueMessageHandlerFactory`.

**Comment by fastnsilver** [comment]
Had to do some backflips to get this working for my purposes.  Sharing this [gist](https://gist.github.com/fastnsilver/20209f7bcdeaa5084b72a5e8857b66fd), in case anyone wants to see.  It's kind of an ugly hack because I had to employ a post processor in order to get argument resolvers to "stick" in `QueueMessageHandler`.

**Comment by jtcotton63** [comment]
+1

@aemruli what is the status on this? Looks like it was assigned to someone and then they dropped it?

**Comment by aemruli** [comment]
fixed with #332
